test/app_test.rb

```diff
@@ -1,15 +1,191 @@
 require "test_helper"
 
 class AppTest < Minitest::Test
+  def setup
+    @app = Minesweeper_2pl::App.new
+    @mocked_app = Minesweeper_2pl::App.new
+    @mocked_game = Minesweeper_2pl::MockedGame.new
+    @mocked_cli = Minesweeper_2pl::MockedCli.new
+    @mocked_app.game = @mocked_game
+    @mocked_app.cli = @mocked_cli
+    @mocked_app.game.setup(100,100)
+  end
 
   def test_that_it_has_an_app_class
-    app = Minesweeper_2pl::App.new
-    refute_nil app
+    refute_nil @mocked_app
+  end
+
+  def test_that_it_has_a_cli
+    refute_nil @mocked_app.cli
+  end
+
+  def test_that_it_has_a_start_method
+    assert @mocked_app.respond_to?(:start)
+  end
+
+  def test_that_it_has_a_setup_method
+    assert @mocked_app.respond_to?(:setup)
+  end
+
+  def test_that_it_creates_a_new_game
+    refute_nil @mocked_app.game
+  end
+
+  def test_that_it_calls_the_apps_setup_method
```

Comment:
Think of these tests, which are tightly coupled to the structure of your production code, as a symptom of telling rather than asking. What if you asked for values instead of telling these objects what to do? Would it be easier to write assertions in your tests and mock a little less?